src/activity_handlers.cpp

```diff
@@ -2728,6 +2730,50 @@ void activity_handlers::try_sleep_finish( player_activity *act, player *p )
     act->set_to_null();
 }
 
+void activity_handlers::build_do_turn( player_activity *act, player *p )
+{
+    item *con_item = act->targets.front().get_item();
+    // this shouldn't happen, unfinished crafts should not dissappear during activity
+    if( !con_item ) {
+        add_msg( m_bad, "The marker item is no longer there, cancelling construction," );
+        p->cancel_activity();
+        return;
+    }
+    std::vector<construction> list_constructions = get_constructions();
+    const construction &built = list_constructions[act->index];
+
+    // item_counter represents the percent progress relative to the base batch time
+    // stored precise to 5 decimal places ( e.g. 67.32 percent would be stored as 6732000 )
+    const int old_counter = con_item->get_var( "construction_progress", 0 );
```

Comment:
Why not use `con_item->item_counter` here?  It would be a little cleaner than using item vars